### ISSUE TITLE
fswatch v1.13 ignores --monitor parameter and always uses default monitor

### ISSUE THREAD

**Issue by Vanav** [emcrisostomo/fswatch#218]
Function below ignores `type` argument and always returns default monitor.
```
  monitor *monitor_factory::create_monitor(fsw_monitor_type type,
                                           std::vector<std::string> paths,
                                           FSW_EVENT_CALLBACK *callback,
                                           void *context)
  {
    switch (type)
    {
    case system_default_monitor_type:
      return create_default_monitor(paths, callback, context);

    default:
#if defined(HAVE_FSEVENTS_FILE_EVENTS)
      return new fsevents_monitor(paths, callback, context);
#elif defined(HAVE_SYS_EVENT_H)
      return new kqueue_monitor(paths, callback, context);
#elif defined(HAVE_PORT_H)
      return new fen_monitor(paths, callback, context);
#elif defined(HAVE_SYS_INOTIFY_H)
      return new inotify_monitor(paths, callback, context);
#elif defined(HAVE_WINDOWS)
      return new windows_monitor(paths, callback, context);
#else
      return new poll_monitor(paths, callback, context);
#endif
    }
}
```
https://github.com/emcrisostomo/fswatch/blob/1ee493b998181f3048444470c292c418ec4f7f96/libfswatch/src/libfswatch/c%2B%2B/monitor_factory.cpp#L69

**Comment by emcrisostomo** [comment]
Hi @Vanav, you're right. It's going to be fixed ASAP.

**Comment by emcrisostomo** [comment]
Fixed in 1.14.0
